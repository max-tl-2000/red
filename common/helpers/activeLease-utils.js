/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { t } from 'i18next';
import { toMoment } from './moment-utils';

export const getActiveLeaseEndDate = activeLeaseWorkflowData => {
  const leaseData = activeLeaseWorkflowData?.leaseData;

  return leaseData?.computedExtensionEndDate || leaseData?.leaseEndDate;
};

export const enhanceChargesWithAdjustedDates = (activeLeaseWorkflowData, timezone) => {
  const { recurringCharges, concessions } = activeLeaseWorkflowData;
  const formattedRecurringCharges = (recurringCharges || []).map(r => {
    const startDate = toMoment(r.startDate, { timezone }).add(2, 'hours').toISOString();
    const endDate = r.endDate ? toMoment(r.endDate, { timezone }).add(2, 'hours').toISOString() : null;
    return {
      ...r,
      startDate,
      endDate,
    };
  });

  const formattedConcessions = (concessions || []).map(c => {
    const startDate = toMoment(c?.startDate, { timezone }).add(2, 'hours').toISOString();
    const endDate = c.endDate ? toMoment(c.endDate, { timezone }).add(2, 'hours').toISOString() : null;
    return {
      ...c,
      startDate,
      endDate,
    };
  });

  return {
    ...activeLeaseWorkflowData,
    recurringCharges: formattedRecurringCharges,
    concessions: formattedConcessions,
  };
};

export const getFormattedLeaseEndDate = (formattedEndDate, daysBeforeLeaseEnds) => {
  const endDate = t('DATE', { date: formattedEndDate });

  if (daysBeforeLeaseEnds < -1) return `${endDate} ${t('DAYS_AGO', { noOfDays: -daysBeforeLeaseEnds })}`;
  if (daysBeforeLeaseEnds === -1) return `${endDate} ${t('REMAINING_YESTERDAY')}`;
  if (daysBeforeLeaseEnds === 0) return `${endDate} ${t('REMAINING_TODAY')}`;
  if (daysBeforeLeaseEnds === 1) return `${endDate} ${t('REMAINING_TOMORROW')}`;
  if (daysBeforeLeaseEnds > 1) return `${endDate} ${t('REMAINING_DAYS', { noOfDays: daysBeforeLeaseEnds })}`;
  return `${endDate}`;
};
