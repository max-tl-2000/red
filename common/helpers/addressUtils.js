/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { formatInventoryName } from '../inventory-helper';
import { isString } from './type-of';
const ZIP_CODE_SEPARATOR = ' ';

export const formatSimpleAddress = ({ addressLine1, addressLine2, city, state, zip, postalCode }) =>
  [addressLine1, addressLine2, city, state, zip || postalCode].filter(item => item).join(' ');

const toReadableString = (values, connector = ', ') => values.filter(item => item).join(connector);
export const formatShortAddress = ({ addressLine1, city }) => [addressLine1, city].filter(item => item).join(', ');

export const formatLongAddress = ({ addressLine1, city, state, zip, postalCode }) =>
  `${[addressLine1, city, state].filter(item => item).join(', ')}${ZIP_CODE_SEPARATOR}${zip || postalCode}`;

export const formatFullAddress = ({ addressLine1, addressLine2, city, state, zip, postalCode }) => ({
  fullAddressLine1: addressLine1,
  fullAddressLine2: addressLine2,
  fullAddressLine3: `${[city, state].filter(item => item).join(', ')}${ZIP_CODE_SEPARATOR}${zip || postalCode}`,
});
export const formatStateAndPostalCode = (state, postalCode) => toReadableString([state, postalCode], ' ');

const hasValidAddress = address => !!Object.keys(address || {}).length;

export const hasValidBuildingAddress = (building = {}) => hasValidAddress(building.address);

export const getUnitAddress = propertyName => (inventory = {}) => {
  if (!hasValidBuildingAddress(inventory.building)) return {};
  const COVE = 'cove';
  const SWPARKME = 'swparkme';
  const SHARON = 'sharon';
  const { city, state, postalCode } = inventory.building.address;
  let addressLine1 = inventory.building.address.addressLine1;
  let addressLine2 = inventory.building.address.addressLine2;

  if (propertyName === SHARON) {
    // sharon = <building.addressLine1>, <building.addressLine2>, <inventory.type> <building.name>-<inventory.name>, <building.city>, <building.state> <building.postalCode>""
    // For example - 350 Sharon Park Drive, Unit G-005, Menlo Park, CA 94025"
    addressLine2 = toReadableString([addressLine2, formatInventoryName(inventory, true)]);
  } else if (propertyName === COVE || propertyName === SWPARKME) {
    // <inventory.address>, <building.city>, <building.state> <building.postalCode>
    // For example -  111 Barbaree Way, Belvedere Tiburon, CA 94920
    addressLine1 = inventory.address;
    addressLine2 = '';
  } else {
    // lark, wood, shore = <building.addressLine1>, <building.addressLine2>, <inventory.type> <inventory.name>, <building.city>, <building.state> <building.postalCode>
    // For example - 2300 Lincoln village circle, Unit 164, Larkspur, CA 94536
    addressLine2 = toReadableString([addressLine2, formatInventoryName(inventory)]);
  }
  return {
    addressLine1,
    addressLine2,
    city,
    state,
    postalCode,
  };
};

export const formatUnitAddress = (inventory = {}) => {
  const { addressLine1: buildingAddressLine1, addressLine2: buildingAddressLine2, city: buildingCity, state: buildingState, postalCode: buildingPostalCode } =
    (inventory.building || {}).address || {};

  const { addressLine1: propertyAddressLine1, addressLine2: propertyAddressLine2, city: propertyCity, state: propertyState, postalCode: propertyPostalCode } =
    (inventory.property || {}).address || {};

  let address;
  if (inventory.inventoryAddress || (inventory.address && isString(inventory.address))) {
    // <inventory.address>, <building.city>, <building.state>, <building.postalCode>
    // eg. 111 Barbaree Way, Belvedere Tiburon, CA 94920
    const inventoryCity = buildingCity || propertyCity;
    const inventoryState = buildingState || propertyState;
    const inventoryPostalCode = buildingPostalCode || propertyPostalCode;

    address = [inventory.inventoryAddress || inventory.address, inventoryCity, formatStateAndPostalCode(inventoryState, inventoryPostalCode)];
  } else if (hasValidBuildingAddress(inventory.building)) {
    // <building.addressLine1>, <building.addressLine2>, <inventory.type> <inventory.name>, <building.city>, <building.state> <building.postalCode>
    // eg. 2300 Lincoln village circle, Unit 164, Larkspur, CA 94536
    address = [
      buildingAddressLine1,
      buildingAddressLine2,
      formatInventoryName(inventory),
      buildingCity,
      formatStateAndPostalCode(buildingState, buildingPostalCode),
    ];
  }

  // <property.addressLine1>, <property.addressLine2>, <inventory.type> <inventory.name>, <property.city>, <property.state> <property.postalCode>
  // eg. 2300 Lincoln village circle, Unit 164, Larkspur, CA 94536
  if (!address) {
    address = [
      propertyAddressLine1,
      propertyAddressLine2,
      formatInventoryName(inventory),
      propertyCity,
      formatStateAndPostalCode(propertyState, propertyPostalCode),
    ];
  }

  return toReadableString(address);
};

export const formatPropertyAddress = ({ addressLine1, addressLine2, city, state, postalCode }) =>
  toReadableString([addressLine1, addressLine2, city, formatStateAndPostalCode(state, postalCode)]);

export const formatBuildingAddress = inventory => {
  const buildingAddress = inventory.building.address;
  const propertyAddress = inventory.property.address;

  if (!hasValidBuildingAddress(inventory.building)) return formatPropertyAddress(propertyAddress);

  const { addressLine1, addressLine2 } = buildingAddress;
  const city = buildingAddress.city || propertyAddress.city;
  const state = buildingAddress.state || propertyAddress.state;
  const postalCode = buildingAddress.postalCode || propertyAddress.postalCode;

  return toReadableString([addressLine1, addressLine2, city, formatStateAndPostalCode(state, postalCode)]);
};
