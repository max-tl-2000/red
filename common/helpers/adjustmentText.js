/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { t } from 'i18next';
import { getFixedAmount } from './number';
import { formatMoney } from '../money-formatter';
import { USD } from '../currency';
import { DALTypes } from '../enums/DALTypes';

const shortPeriods = {
  year: 'y',
  week: 'w',
  month: 'm',
  day: 'd',
  hour: 'h',
};

const calculateAdjustmentValue = concession => {
  const { absoluteAdjustment, relativeAmount, amountVariableAdjustment, floorCeilingAmount } = concession;

  let adjustment = 0;

  if (amountVariableAdjustment > 0) {
    adjustment = amountVariableAdjustment;
  } else if (floorCeilingAmount > 0) {
    adjustment = floorCeilingAmount;
  } else if (relativeAmount) {
    adjustment = relativeAmount;
  } else if (parseFloat(absoluteAdjustment) !== 0) {
    adjustment = Math.abs(absoluteAdjustment);
  }

  return getFixedAmount(adjustment, 2);
};

const getShortPeriodFor = period => (period && shortPeriods[period]) || period;

export const adjustmentText = (concession, term) => {
  // *** This is where we need the override value
  const { recurringCount } = concession;

  const shortPeriod = getShortPeriodFor(term.period);
  const adjustmentValue = calculateAdjustmentValue(concession);
  const { result: formattedAdjustmentValue } = formatMoney({
    amount: adjustmentValue,
    currency: USD.code,
  });
  const adjustment = `(${formattedAdjustmentValue}${t('PER_PERIOD', {
    period: shortPeriod,
  })}`;

  const count = recurringCount > 0 ? recurringCount : term.termLength;

  return `${adjustment} ${t('FOR_PERIOD', {
    length: count,
    period: shortPeriod,
  })})`;
};

export const getStartingAtPriceText = (marketRent, term = { termLength: 12, period: DALTypes.LeasePeriod.MONTH }, appendPeriodInfo = true) => {
  const { integerPart, decimalPart } = formatMoney({
    amount: marketRent,
    currency: USD.code,
  });
  const startingAtPriceText = t('STARTING_AT', {
    integerPart,
    decimalPart,
  });
  if (!appendPeriodInfo) return startingAtPriceText;

  const shortPeriod = getShortPeriodFor(term.period);
  const shortPreiordText = t('PER_PERIOD', { period: shortPeriod });
  const forPeriodText = `${shortPreiordText} ${t('FOR_PERIOD', {
    length: term.termLength,
    period: shortPeriod,
  })}`;

  return `${startingAtPriceText}${forPeriodText}`;
};
