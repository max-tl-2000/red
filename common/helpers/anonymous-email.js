/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

const anonymizedEmails = {
  'zlead.co': [/^.{26}@zlead.co$/],
  'reply.craigslist.org': [/^.{32}@reply.craigslist.org$/],
  'renter.apartmentlist.com': [/^.+\..+\..{5}@renter.apartmentlist.com$/],
  'messaging.yelp.com': [/^reply\+([^@]*)@messaging.yelp.com$/],
  'message.my.apartmentguide.com': [/^reply-([^@]*)@message.my.apartmentguide.com$/],
  'email.rent.com': [/^reply-([^@]*)@email.rent.com$/],
  'abodoapts.com': [/^abodoleads\+([^@]*)@abodoapts.com$/],
};

const isAnonymizedEmail = (email, searchExpressions = []) => searchExpressions.some(expression => new RegExp(expression, 'i').test(email));

export const isAnonymousEmail = email => email && Object.keys(anonymizedEmails).some(provider => isAnonymizedEmail(email, anonymizedEmails[provider]));
