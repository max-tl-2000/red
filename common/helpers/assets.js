/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

const getUrlParams = ({ tenantName, cloudEnv }, apiToken) => `env=${cloudEnv}&tenant=${tenantName}&api-token=${apiToken}`;

export const formatStaticAssetUrl = (ctx, assetName, assetType) => {
  const { isPublicEnv, domainSuffix, reverseProxyUrl, rpImageToken, cloudEnv } = ctx;

  if (!domainSuffix || !reverseProxyUrl || !rpImageToken || !cloudEnv || !assetName || !assetType) return '';

  const staticAssetsBaseUrl = isPublicEnv ? `https://static.${domainSuffix}/images` : `${reverseProxyUrl}/images`;
  const assetUrl = `${staticAssetsBaseUrl}/${assetType.toLowerCase()}/${assetName}`;

  return isPublicEnv ? assetUrl : `${assetUrl}?${getUrlParams({ cloudEnv, tenantName: 'static' }, rpImageToken)}`;
};
