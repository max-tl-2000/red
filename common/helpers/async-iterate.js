/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

export const asyncIterate = (items, { itemCb, done, delay = 1000 } = {}) => {
  if (!Array.isArray(items)) {
    throw new Error('first argument must be an array');
  }
  if (typeof itemCb !== 'function') {
    throw new Error('itemCb callback is required');
  }

  const len = items.length;
  let index = 0;

  if (len === 0) {
    done && done();
    return;
  }

  const process = () => {
    const item = items[index];
    itemCb({ item, index }, () => {
      index++;
      if (index === len) {
        done && done();
        return;
      }
      setTimeout(process, delay);
    });
  };
  process();
};

export const iterateOverArray = (items = [], args) => {
  const { chunkSize = 100, onChunk, done } = args;
  const { length } = items;

  const totalGroups = Math.ceil(length / chunkSize);
  let processed = 0;
  let cancelled = false;
  let iterationDone = false;

  const process = () => {
    if (cancelled) {
      return;
    }

    if (processed === totalGroups) {
      iterationDone = true;
      done && done();
      return;
    }

    const from = processed * chunkSize;
    const to = from + chunkSize;
    const arr = items.slice(from, to);

    setTimeout(() => {
      if (cancelled) return;
      onChunk &&
        onChunk({ arr, from, to: to - 1 }, () => {
          processed++;
          setTimeout(process);
        });
    });
  };

  process();

  return () => {
    if (iterationDone || cancelled) return;
    cancelled = true;
  };
};
