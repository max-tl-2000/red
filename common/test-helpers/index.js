/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
import { unmountComponentAtNode } from 'react-dom';
import when from './when';
import and from './and';
import * as overrider from './overrider';
import * as sandboxer from './sandboxer';
import sleep from '../helpers/sleep';

sandboxer.setSinon(sinon);

const expect = chai.expect;
const assert = chai.assert;

chai.use(sinonChai);

let div;

if (global.document) {
  beforeEach(() => {
    div = document.createElement('div');
    div.setAttribute('id', 'sandbox');

    document.body.appendChild(div);
  });
}

afterEach(() => {
  if (global.document && div) {
    unmountComponentAtNode(div);
    document.body.removeChild(div);
    div = null;
  }

  overrider.restore();
  sandboxer.restore();
});

const getSandBoxDiv = () => div;

// helper method to trigger a real click event in the browser
// jquery.fn.click is not the same as this as the event need
// to also generate the capture phase
const triggerMouseEvent = (ele, evt) => {
  const mEvent = document.createEvent('MouseEvent');
  // interface
  // event.initMouseEvent(type, canBubble, cancelable, view,
  //                    detail, screenX, screenY, clientX, clientY,
  //                    ctrlKey, altKey, shiftKey, metaKey,
  //                    button, relatedTarget);
  mEvent.initMouseEvent(evt, true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);

  ele.dispatchEvent(mEvent);
};

export { getSandBoxDiv, triggerMouseEvent, overrider, expect, assert, sinon, when, sandboxer, and, sleep };
