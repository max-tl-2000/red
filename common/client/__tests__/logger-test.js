/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

/* eslint-disable global-require */
import { override, restore } from 'test-helpers/overrider';
import { deferred } from '../../helpers/deferred';
const { mockModules } = require('test-helpers/mocker').default(jest);

describe('client/logger', () => {
  beforeEach(() => {
    mockModules({
      '../../helpers/globals': {
        window: {
          navigator: { userAgent: 'Mock User Agent' },
          location: { href: 'https://mock/location/to/resource' },
        },
      },
    });

    override(global.console, {
      log: () => {},
      debug: () => {},
      error: () => {},
      info: () => {},
      warn: () => {},
    });
  });

  afterEach(() => {
    jest.resetModules();
    restore();
  });

  describe('logger', () => {
    it('should not fail if used without initialization', () => {
      const mockConsole = { warn: jest.fn() };

      override(global.console, mockConsole);

      const { logger } = require('../logger');

      logger.trace('some method called 1');
      expect(mockConsole.warn).toHaveBeenCalledWith('logger not initialized', 'some method called 1');

      logger.error('some method called 2');
      expect(mockConsole.warn).toHaveBeenCalledWith('logger not initialized', 'some method called 2');

      logger.info('some method called 3');
      expect(mockConsole.warn).toHaveBeenCalledWith('logger not initialized', 'some method called 3');

      logger.warn('some method called 4');
      expect(mockConsole.warn).toHaveBeenCalledWith('logger not initialized', 'some method called 4');

      logger.debug('some method called 5');
      expect(mockConsole.warn).toHaveBeenCalledWith('logger not initialized', 'some method called 5');
    });
  });

  describe('initClientLogger', () => {
    it('should throw if not apiClient provided', () => {
      const { initClientLogger } = require('../logger');

      expect(() => initClientLogger()).toThrowError('apiClient parameter not provided');
    });

    it('should use the console object in development mode as well as the api endpoint only for error messages', () => {
      const { logger, initClientLogger } = require('../logger');

      const consoleMock = {
        trace: jest.fn(),
        debug: jest.fn(),
        info: jest.fn(),
        error: jest.fn(),
        warn: jest.fn(),
      };

      override(global.console, consoleMock, /* allowNonExistent */ true);
      override(process.env, { NODE_ENV: 'development' });

      const apiClient = { post: jest.fn(() => Promise.resolve()) };

      initClientLogger({ apiClient });

      logger.trace('a trace message');
      logger.debug('a debug message');
      logger.info('an info message');
      logger.error('an error message');
      logger.warn('a warn message');

      expect(consoleMock.trace.mock.calls).toMatchSnapshot();
      expect(consoleMock.debug.mock.calls).toMatchSnapshot();
      expect(consoleMock.info.mock.calls).toMatchSnapshot();
      expect(consoleMock.error.mock.calls).toMatchSnapshot();
      expect(consoleMock.warn.mock.calls).toMatchSnapshot();

      expect(apiClient.post).toHaveBeenCalledTimes(1);

      const { data } = apiClient.post.mock.calls[0][1];

      expect(data[0].severity).toEqual('error');
      expect(data[0].loggingMessage).toEqual('an error message');
    });

    it('should do a post to the provided client api only for error severity', () => {
      const { logger, initClientLogger } = require('../logger');

      const apiClient = {
        post: jest.fn(() => Promise.resolve()),
      };

      initClientLogger({ apiClient });

      logger.trace('some method called 1');
      logger.debug('some method called 2');
      logger.warn('some method called 3');
      logger.info('some method called 4');
      logger.error('some methd called 5');

      expect(apiClient.post).toHaveBeenCalledTimes(1);
      expect(apiClient.post.mock.calls).toMatchSnapshot();
    });

    it('should also accept a extra object to be sent to the api endpoint in case of severity = error', () => {
      const { logger, initClientLogger } = require('../logger');

      const apiClient = {
        post: jest.fn(() => Promise.resolve()),
      };

      initClientLogger({ apiClient });

      logger.error({ custom: 'value', to: 'pass', for: 'logger' }, 'some method called 1');

      expect(apiClient.post).toHaveBeenCalledTimes(1);
      expect(apiClient.post.mock.calls).toMatchSnapshot();
    });

    it('failures to post messages should be logged as errors', async () => {
      const mockConsole = { error: jest.fn() };
      const d = deferred();

      override(global.console, mockConsole);

      const { logger, initClientLogger } = require('../logger');

      const apiClient = {
        post: jest.fn(() => d),
      };

      initClientLogger({ apiClient });

      logger.error({ custom: 'value', to: 'pass', for: 'logger' }, 'some method called 1');

      d.reject({ status: 500 }); // this object simulates a network error

      try {
        await d;
      } catch (ex) {
        // no needed to handle this during testing
      }

      expect(mockConsole.error).toHaveBeenCalled();
      expect(mockConsole.error.mock.calls).toMatchSnapshot();
    });

    it('should allow to add the user info', () => {
      const { logger, initClientLogger } = require('../logger');

      const apiClient = {
        post: jest.fn(() => Promise.resolve()),
      };

      initClientLogger({
        apiClient,
        getContextData: () =>
          // this function can return any object
          // and that will just be added as the contextData object
          // to the logger
          ({
            userId: 1,
            userName: 'Some name',
          }),
      });

      logger.error({ custom: 'value', to: 'pass', for: 'logger' }, 'some method called 1');

      expect(apiClient.post).toHaveBeenCalledTimes(1);
      expect(apiClient.post.mock.calls).toMatchSnapshot();
    });
  });
});
