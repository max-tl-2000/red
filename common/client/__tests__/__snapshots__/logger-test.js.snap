// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client/logger initClientLogger failures to post messages should be logged as errors 1`] = `
Array [
  Array [
    "some method called 1, {
  \\"severity\\": \\"error\\",
  \\"userAgent\\": \\"Mock User Agent\\",
  \\"currentUrl\\": \\"https://mock/location/to/resource\\",
  \\"custom\\": \\"value\\",
  \\"to\\": \\"pass\\",
  \\"for\\": \\"logger\\"
}",
  ],
  Array [
    "UNABLE TO POST LOG!",
    Object {
      "status": 500,
    },
  ],
]
`;

exports[`client/logger initClientLogger should allow to add the user info 1`] = `
Array [
  Array [
    "/log",
    Object {
      "data": Array [
        Object {
          "contextData": Object {
            "userId": 1,
            "userName": "Some name",
          },
          "currentUrl": "https://mock/location/to/resource",
          "custom": "value",
          "for": "logger",
          "loggingMessage": "some method called 1",
          "severity": "error",
          "to": "pass",
          "userAgent": "Mock User Agent",
        },
      ],
    },
  ],
]
`;

exports[`client/logger initClientLogger should also accept a extra object to be sent to the api endpoint in case of severity = error 1`] = `
Array [
  Array [
    "/log",
    Object {
      "data": Array [
        Object {
          "currentUrl": "https://mock/location/to/resource",
          "custom": "value",
          "for": "logger",
          "loggingMessage": "some method called 1",
          "severity": "error",
          "to": "pass",
          "userAgent": "Mock User Agent",
        },
      ],
    },
  ],
]
`;

exports[`client/logger initClientLogger should do a post to the provided client api only for error severity 1`] = `
Array [
  Array [
    "/log",
    Object {
      "data": Array [
        Object {
          "currentUrl": "https://mock/location/to/resource",
          "loggingMessage": "some methd called 5",
          "severity": "error",
          "userAgent": "Mock User Agent",
        },
      ],
    },
  ],
]
`;

exports[`client/logger initClientLogger should use the console object in development mode as well as the api endpoint only for error messages 1`] = `
Array [
  Array [
    "a trace message",
    Object {},
  ],
]
`;

exports[`client/logger initClientLogger should use the console object in development mode as well as the api endpoint only for error messages 2`] = `
Array [
  Array [
    "a debug message",
    Object {},
  ],
]
`;

exports[`client/logger initClientLogger should use the console object in development mode as well as the api endpoint only for error messages 3`] = `
Array [
  Array [
    "an info message",
    Object {},
  ],
]
`;

exports[`client/logger initClientLogger should use the console object in development mode as well as the api endpoint only for error messages 4`] = `
Array [
  Array [
    "an error message",
    Object {},
  ],
]
`;

exports[`client/logger initClientLogger should use the console object in development mode as well as the api endpoint only for error messages 5`] = `
Array [
  Array [
    "a warn message",
    Object {},
  ],
]
`;
